app/services/fetch_remote_status_service.rb

```diff
@@ -40,6 +40,6 @@ def process_atom(url, prefetched_body:)
   end
 
   def confirmed_domain?(domain, account)
-    account.domain.nil? || domain.casecmp(account.domain).zero? || domain.casecmp(Addressable::URI.parse(account.remote_url || account.uri).normalized_host).zero?
+    account.domain.nil? || domain.casecmp(account.domain).zero? || domain.casecmp(Addressable::URI.parse(account.remote_url.blank? ? account.uri : account.remote_url).normalized_host).zero?
```

Comment:
`account.remote_url.presence || account.uri`

Reply:
Didn't know that one, nice! :)
Changed